src/lib/reasoners/th_util.mli

```diff
@@ -30,13 +30,28 @@
 
 type answer = (Explanation.t * Expr.Set.t list) option
 
-
 type theory =
   | Th_arith
   | Th_sum
   | Th_adt
   | Th_arrays
   | Th_UF
+[@@deriving show]
+
+type limit_kind =
+  | Above
+  | Below
+  | Exact
+  (* Type used to discriminate between limits from above or below. The
+     [Exact] constructor means the value isn't a limit but an exact value. *)
```

Comment:
I think the best ADT would be:
```ocaml
type limit_kind =
| Above
| Below 

type 'a optimized_split_value =
| Minfinity
| Pinfinity
| Value of 'a 
| Limit of limit_kind * 'a
```
I don't like `Exact` or `None`. I agree `None` is worse than `Exact` but `Exact` doesn't mean anything for a limit.

Reply:
That is because you think of it as a limit whereas I think of it as a bound on the set of possible values, in which case an "exact bound" makes sense :) But that ADT is also fine, yes.